src/dattobd.c

```diff
@@ -2883,11 +2883,6 @@ static void __on_bio_read_complete(struct bio *bio, int err){
 		goto error;
 	}
 
-	for(i = 0; i < bio->bi_vcnt; i++){
-		bio->bi_io_vec[i].bv_len = PAGE_SIZE;
-		bio->bi_io_vec[i].bv_offset = 0;
```

Comment:
so does somebody set these to the correct value each time on the way in? Are you sure this is true for older kernels?
you might wanna make a feature test for this and only skip it if the kernel supports multipage bios.
Always defer to tom, but I'm guessing this reset was necessary for older kernels?

Reply:
Spoke to Tom, has to do with pre-bvec-iterator kernels. On those versions, reading causes `offset` to increase and `len` to decrease. This loop resets those positions before we pass it on. After bvec iterators (3.14), it's unnecessary. Will fix.